### PR TITLE
25259 context root fat

### LINKED ISSUE #25259: Feature Test Summary - for LTPA useContextRootAsCookiePath
## Test Strategy

**Describe the test strategy & approach for this feature, and describe how the approach verifies the functions delivered by this feature.**

_For any feature, be aware that only FAT tests (not unit or BVT) are executed in our cross platform testing. To ensure cross platform testing ensure you have sufficient FAT coverage to verify the feature._

_If delivering tests outside of the standard Liberty FAT framework, do the tests push the results into cognitive testing database (if not, consult with the CSI Team who can provide advice and verify if results are being received)?_

### List of FAT projects affected

*  new FAT project: _dev/com.ibm.ws.security.token.ltpa_fat/_
*  Existing FAT project: _dev/com.ibm.ws.security.jwtsso_fat/_

### Test strategy

* What functionality is new or modified by this feature?
   * Use the context root as the LTPA/JWT cookie path to use separate LTPA/JWT tokens for different applications (#16235).

* What are the positive and negative tests for that functionality? (Tell me the specific scenarios you tested. What kind of tests do you have for when everything ends up working (positive tests)? What about tests that verify we fail gracefully when things go wrong (negative tests)? See the [Positive and negative tests](https://github.ibm.com/websphere/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)#positive-and-negative-tests) section of the Feature Test Summary Process wiki for more detail.)

**Automated Testcases:**

###  new FAT project: _dev/com.ibm.ws.security.token.ltpa_fat/_
1. `testCookiePath_useContextRootForSSOCookiePath_true`

2. `testCookiePath_useContextRootForSSOCookiePath_false`

3. `testCookiePath_2apps_useContextRootForSSOCookiePath_true_basicauth`

4. `testCookiePath_2apps_useContextRootForSSOCookiePath_false_basicauth`

5. `testCookiePath_2apps_useContextRootForSSOCookiePath_true_formlogin`

6. `testCookiePath_2apps_useContextRootForSSOCookiePath_false_formlogin`

7. `testCookiePath_useContextRootForSSOCookiePath_true_formlogout`

8. `testCookiePath_useContextRootForSSOCookiePath_false_formlogout`

9. `testCookiePath_useContextRootForSSOCookiePath_true_ProgrammaticAPI`

10. `testCookiePath_useContextRootForSSOCookiePath_false_ProgrammaticAPI`

11. `testServerConfiguration_useContextRootForSSOCookiePath_badValue`

###  Existing FAT project: _dev/com.ibm.ws.security.jwtsso_fat/_

1. `test_obtainLtpa_setContextRoot_reconfigureJwtSso_reaccessWithLtpaCookie`
2. `test_obtainLtpa_setContextRoot_reconfigureJwtSso_accessSecondAppWithLtpaCookie`
3. `test_obtainJwt_accessNewProtectedResource_contextRootSetInSSOCookie`
4. `test_obtainJwt_accessNewProtectedResource_noContextRootInSSOCookie`

* What manual tests are there (if any)? (Note: Automated testing is expected for all features with manual testing considered an exception to the rule.)

## Confidence Level

**Collectively as a team you need to assess your confidence in the testing delivered based on the values below.  This should be done as a team and not an individual to ensure more eyes are on it and that pressures to deliver quickly are absorbed by the team as a whole.**

Please indicate your confidence in the testing (up to and including FAT) delivered with this feature by selecting one of these values:

0 - No automated testing delivered

1 - We have minimal automated coverage of the feature including golden paths.  There is a relatively high risk that defects or issues could be found in this feature.

2 - We have delivered a reasonable automated coverage of the golden paths of this feature but are aware of gaps and extra testing that could be done here.  Error/outlying scenarios are not really covered.  There are likely risks that issues may exist in the golden paths

3 - We have delivered all automated testing we believe is needed for the golden paths of this feature and minimal coverage of the error/outlying scenarios.  There is a risk when the feature is used outside the golden paths however we are confident on the golden path.  Note:  This may still be a valid end state for a feature... things like Beta features may well suffice at this level.

**4 - We have delivered all automated testing we believe is needed for the golden paths of this feature and have good coverage of the error/outlying scenarios.  While more testing of the error/outlying scenarios could be added we believe there is minimal risk here and the cost of providing these is considered higher than the benefit they would provide.**

5 - We have delivered all automated testing we believe is needed for this feature.  The testing covers all golden path cases as well as all the error/outlying scenarios that make sense.  We are not aware of any gaps in the testing at this time. No manual testing is required to verify this feature.

Based on your answer above, for any answer other than a 4 or 5 please provide details of what drove your answer.  Please be aware, it may be perfectly reasonable in some scenarios to deliver with any value above.  We may accept no automated testing is needed for some features, we may be happy with low levels of testing on samples for instance so please don't feel the need to drive to a 5.  We need your honest assessment as a team and the reasoning for why you believe shipping at that level is valid. What are the gaps, what is the risk etc.  Please also provide links to the follow on work that is needed to close the gaps (should you deem it needed)

4 - We have delivered all automated testing we believe is needed for the golden paths of this feature and have good coverage of the error/outlying scenarios. While more testing of the error/outlying scenarios could be added we believe there is minimal risk here and the cost of providing these is considered higher than the benefit they would provide.

### PR DESCRIPTION
**Pull request by mshah0722**
What we're testing: `<webAppSecurity useContextRootForSSOCookiePath="true" />`
- New Server Configuration attribute: _useContextRootForSSOCookiePath_

We created a new FAT project called: `dev/com.ibm.ws.security.token.ltpa_fat/`

We created a new testClass called: `ContextRootCookiePathTests.java`

We created 7 new testCases:
1. testCookiePath_useContextRootForSSOCookiePath_true
2. testCookiePath_useContextRootForSSOCookiePath_false  
3. testCookiePath_2apps_useContextRootForSSOCookiePath_true_basicauth
4. testCookiePath_2apps_useContextRootForSSOCookiePath_false_basicauth  
5. testCookiePath_2apps_useContextRootForSSOCookiePath_true_formlogin
6. testCookiePath_2apps_useContextRootForSSOCookiePath_false_formlogin  
7. testServerConfiguration_useContextRootForSSOCookiePath_badValue  

TODO: Enable tests and repeat classRule after https://github.com/OpenLiberty/open-liberty/pull/24876 is delivered

### CODE REVIEW

File: dev/com.ibm.ws.security.token.ltpa_fat/fat/src/com/ibm/ws/security/token/ltpa/fat/BasicAuthLTPATests.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        Log.info(thisClass, "testSuccessfulLoginWithCookieForSameContextRoot", "Cookie: " + cookie);
+        Log.info(thisClass, "testSuccessfulLoginWithCookieForSameContextRoot", "Cookie Path: " + cookiePath);
+
+    }
+
+    /**
+     * Verify the following:
+     * <OL>
+     * <LI>Set useContextRootForSSOCookiePath to true
+     * <LI>Attempt to access a simple servlet configured for basic auth1 with a valid userId (mnagerUser) and password.
```
Comment (Zech-Hein) [inline dev/com.ibm.ws.security.token.ltpa_fat/fat/src/com/ibm/ws/security/token/ltpa/fat/BasicAuthLTPATests.java]:
```suggestion
     * <LI>Attempt to access app1 (basicauth1/SimpleServlet) with a valid userId (mnagerUser) and password.
```

**Review by Zech-Hein — APPROVED** [review]

**Comment by mshah0722** [comment]
#libby

**Comment by mshah0722** [comment]
#build

**Comment by mshah0722** [comment]
The collective repository FAT failures are unrelated to this PR. The rest of the PB results look good.

**Pull request merged**